### PR TITLE
test  case for core

### PR DESCRIPTION
**Pull request by tomdong2010**

### CODE REVIEW

File: account/branch_test.go [hunk 1]
```diff
@@ -95,3 +96,19 @@ func TestBranch(t *testing.T) {
 	}
 	fmt.Println("TestBranch:success test num:", count)
 }
+func TestCreatBranch(t *testing.T) {
```
Comment (pingke) [inline account/branch_test.go]: add empty line

File: core/types/rootblock_test.go [hunk 2]
```diff
@@ -88,6 +89,33 @@ func TestRootBlockEncoding(t *testing.T) {
 	check("serialize", common.Bytes2Hex(bytes), common.Bytes2Hex(blockEnc))
 
 }
+func TestDataSize(t *testing.T) {
```
Comment (pingke) [inline core/types/rootblock_test.go]: add empty line

File: core/types/transaction_test.go [hunk 3]
```diff
@@ -3,8 +3,12 @@ package types
 import (
 	"bytes"
 	"crypto/ecdsa"
+	"encoding/hex"
 	"github.com/QuarkChain/goquarkchain/account"
+	"github.com/QuarkChain/goquarkchain/serialize"
```
Comment (pingke) [inline core/types/transaction_test.go]: move below

File: core/types/transaction_test.go [hunk 4]
```diff
@@ -185,3 +189,81 @@ func TestTransactionPriceNonceSort(t *testing.T) {
...
+		EvmTx:  evmTx2,
+		TxType: EvmTx,
+	}
+	txBytes2, err := serialize.SerializeToBytes(&tx2)
+	if err != nil {
+		t.Fatal("Serialize error: ", err)
+	}
+
+	check("EvmTransaction min len", len(txBytes), 120)
+	check("EvmTransaction max len", len(txBytes2), 162)
```
Comment (tomdong2010) [inline core/types/transaction_test.go]: this was not same as python 's 258

File: account/branch_test.go (hunk 1 above)
Comment (tomdong2010) [reply account/branch_test.go]: ok

File: core/types/rootblock_test.go (hunk 2 above)
Comment (tomdong2010) [reply core/types/rootblock_test.go]: ok

File: core/types/transaction_test.go (hunk 3 above)
Comment (tomdong2010) [reply core/types/transaction_test.go]: ok

File: core/types/rootblock.go [hunk 5]
```diff
@@ -76,9 +77,35 @@ func (b *RootBlockHeader) GetCoinbaseAmount() *TokenBalances {
...
+
+	if b.Signature != [65]byte{} {
+		return true
+	} else {
+		return false
+	}
+
+}
+
+func (b *RootBlockHeader) VerifySignature() bool {
```
Comment (pingke) [inline core/types/rootblock.go]: why?

File: core/types/rootblock.go [hunk 6]
```diff
@@ -76,9 +77,35 @@ func (b *RootBlockHeader) GetCoinbaseAmount() *TokenBalances {
...
+	return true
+}
+
+func (b *RootBlockHeader) GetHashForMining() (hash common.Hash) {
```
Comment (pingke) [inline core/types/rootblock.go]: no need

File: core/types/rootblock.go [hunk 7]
```diff
@@ -76,9 +77,35 @@ func (b *RootBlockHeader) GetCoinbaseAmount() *TokenBalances {
 	return NewEmptyTokenBalances()
 }
 
+func (b *RootBlockHeader) IsSigned() bool {
```
Comment (pingke) [inline core/types/rootblock.go]: no needed

File: core/types/rootblock.go (hunk 5 above)
Comment (tomdong2010) [reply core/types/rootblock.go]: delete it

File: core/types/transaction_test.go (hunk 4 above)
Comment (tomdong2010) [reply core/types/transaction_test.go]: now  210

File: core/types/rootblock.go (hunk 5 above)
Comment (tomdong2010) [reply core/types/rootblock.go]: I changed it recently

**Pull request merged**